charts/vineyard/templates/master/service.yaml

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020-2023 Alibaba Group Holding Limited.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+{{- $clientPort := int .Values.master.ports.client }}
+{{- $peerPort := int .Values.master.ports.peer }}
+{{- $name := include "vineyard.name" . }}
+{{- $fullName := include "vineyard.fullname" . }}
+{{- $chart := include "vineyard.chart" . }}
+
+{{- if not .Values.master.externalEndpoint }}
```

Comment:
Could you provide the reason of providing both service and headless service reason for master?

Reply:
Is it imperative to use different ways to access master pods from masters and workers? According to my knowledge, Kubernetes API Server connects multiple endpoints.

Reply:
Thanks, I will try to use the headless service only.